types/precison_timestamp_types.go

```diff
@@ -9,20 +9,19 @@ import (
 // TimePrecision is used to represent the precision of a timestamp
 type TimePrecision int32
 
-const PrecisionUnknown TimePrecision = -1
-
-// precision values are proto values
 const (
-	PrecisionSeconds TimePrecision = iota
-	PrecisionDeciSeconds
-	PrecisionCentiSeconds
-	PrecisionMilliSeconds
-	PrecisionEMinus4Seconds // 10^-4 of seconds
-	PrecisionEMinus5Seconds // 10^-5 of seconds
-	PrecisionMicroSeconds
-	PrecisionEMinus7Seconds // 10^-7 of seconds
-	PrecisionEMinus8Seconds // 10^-8 of seconds
-	PrecisionNanoSeconds
+	PrecisionUnknown TimePrecision = -1
+	// below precision values are proto values
+	PrecisionSeconds        TimePrecision = 0
+	PrecisionDeciSeconds    TimePrecision = 1
+	PrecisionCentiSeconds   TimePrecision = 2
+	PrecisionMilliSeconds   TimePrecision = 3
+	PrecisionEMinus4Seconds TimePrecision = 4 // 10^-4 of seconds
+	PrecisionEMinus5Seconds TimePrecision = 5 // 10^-5 of seconds
+	PrecisionMicroSeconds   TimePrecision = 6
+	PrecisionEMinus7Seconds TimePrecision = 7 // 10^-7 of seconds
+	PrecisionEMinus8Seconds TimePrecision = 8 // 10^-8 of seconds
+	PrecisionNanoSeconds    TimePrecision = 9
 )
```

Comment:
why this change?

Reply:
This was a review comment that I failed to submit on the precision patch. It's unrelated to this and should be done separately.

I felt like the implicit iota didn't clarify that the numbers are meaningful. It seemed too easy for a future editor to change things and not realize the importance. Generally, I don't we should iota unless the number is meaningless (just an artifact of go enum pattern) to our code/business logic. I generally follow a similar pattern across languages. For example, in Java, one should store concrete values if they want them and not try to overload ordinal.

Reply:
Raised [this](https://github.com/substrait-io/substrait-go/pull/49) PR to fix it. Removed from this PR